adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpContext.java

```diff
@@ -136,4 +140,24 @@ ProtonDelivery updateDelivery(final DeliveryState state, final boolean settle) {
     boolean isRemotelySettled() {
         return delivery.remotelySettled();
     }
+
+    //---------------------------------------------< private methods >---
+    /**
+     * Creates an ErrorCondition using the given service invocation error to provide an error condition value
+     * and description.
+     *
+     * @param error The service invocation error.
+     * @return The ErrorCondition.
+     */
+    private ErrorCondition getErrorCondition(final ServiceInvocationException error) {
+        switch (error.getErrorCode()) {
+        case HttpURLConnection.HTTP_BAD_REQUEST:
+            return ProtonHelper.condition(Constants.AMQP_BAD_REQUEST, error.getMessage());
+        case HttpURLConnection.HTTP_FORBIDDEN:
+            return ProtonHelper.condition(AmqpError.UNAUTHORIZED_ACCESS, error.getMessage());
+        default:
+            // TODO: we default to null?
+            return null;
```

Comment:
we need to return an ErrorCondition in any case. I am sure you will be able to come up with a reasonable default.

Reply:
ok, i was thinking of using `amqp:precondition-failed` since if there is a client error, then there is something the adapter expects which the client is not providing?

Reply:
makes sense